doc/whats_new/v0.22.rst

```diff
@@ -236,6 +236,13 @@ Changelog
   :user:`Matt Hancock <notmatthancock>` and
   :pr:`5963` by :user:`Pablo Duboue <DrDub>`.
 
+- |Fix| Stacking and Voting estimators now ensure that their underlying
+  estimators are either all classifiers or all regressors.
+  We introduced a new base class
+  :class:`ensemble.base._BaseHeterogeneousEnsemble` to raise consistent error
```

Comment:
Should we include a private class in the whats new? This can be something like:

> Stacking and Voting estimators now raise consistent error messages.

Reply:
I might be misunderstood what @NicolasHug meant by adding a link?
Did you mean mentioning the class or do you expect something else?

Reply:
Since we are not generating a new API doc for `_BaseHeterogeneousEnsemble`, there is nothing to link to: https://76528-843222-gh.circle-artifacts.com/0/doc/whats_new/v0.22.html#sklearn-ensemble

Reply:
The links referred to the Stacking and Voting estimators, sorry if that wasn't clear. I agree we shouldn't link a private class. (and I'm also fine not linking the estimators... it's just a nit)

Reply:
Oh ok make sense.